### PR TITLE
Patch : To handle file names with spaces. 

### PR DESCRIPTION
**Pull request by fahad-israr**
- Add url decoer to handle file names with spaces.
- Resolves https://github.com/starfixdev/starfix/issues/72

### CODE REVIEW

File: cli/src/main/java/dev/starfix/Starfix.java [hunk 1]
```diff
@@ -43,7 +45,12 @@ public int config() throws Exception {
 
     @Command(name = "clone")
     public int cloneCmd(@Parameters(index = "0") String url) {
-
+        try{
+            url = URLDecoder.decode(url,"UTF-8");
```
Comment (maxandersen) [inline cli/src/main/java/dev/starfix/Starfix.java:49]: shouldn't this not only be done for the file path ?

if there are spaces in the git clone part it needs to still be encoded

Comment (fahad-israr) [reply cli/src/main/java/dev/starfix/Starfix.java:49]: Yes true, it should be there in the blob case only. Would move it down to that specific case 👍

Comment (maxandersen) [reply cli/src/main/java/dev/starfix/Starfix.java:49]: should also start adding tests for these changes so we don 't break it in future.

Comment (fahad-israr) [reply cli/src/main/java/dev/starfix/Starfix.java:49]: Yes , will add tests.

**Comment by fahad-israr** [comment]
@maxandersen have made suggested changes already @[commmit](https://github.com/starfixdev/starfix/pull/74/commits/bb364013e88b7082335e7009e3c9716114d09e3e)  . It solves the linked issue .
As far as the tests are concerned I am not sure what tests to add here  and would need suggestion on that and I am opening a separate issue so that we don't  miss that .

**Comment by maxandersen** [comment]
To test the code we need the code to be testable. Atm all the main code is in one big method and thus very hard to write tests for it.

**Comment by fahad-israr** [comment]
> To test the code we need the code to be testable. Atm all the main code is in one big method and thus very hard to write tests for it.

I've opened a separate Issue for adding Tests . I'd try to break down code in different functions (though we still have many ) and probably that could help in testing .

**Review by maxandersen — APPROVED** [review]

**Pull request merged**